Source/santactl/Commands/SNTCommandFileInfo.m

```diff
@@ -355,276 +406,234 @@ - (SNTAttributeBlock)signingChain {
+  // how many file info objects are being outputted.  So both empty and singleton result sets are
+  // still enclosed in brackets.
+  if (self.jsonOutput) printf("[\n");
+
+  NSFileManager *fm = [NSFileManager defaultManager];
+  NSString *cwd = [fm currentDirectoryPath];
+  for (NSString *path in filePaths) {
+    NSString *fullPath = [path stringByStandardizingPath];
+    if ([path characterAtIndex:0] != '/') {
+      fullPath = [cwd stringByAppendingPathComponent:fullPath];
+    }
+    [self recurseAtPath:fullPath];
+  }
 
-  // Wait for all the writes to the outputHashes to finish
-  dispatch_group_wait(outputHashesGroup, DISPATCH_TIME_FOREVER);
+  if (self.jsonOutput) printf("\n]\n"); // print closing bracket of JSON output array
 
-  if (outputHashes.count) [self printOutputHashes:outputHashes];
   exit(0);
 }
 
-#pragma mark FileInfo helper methods
-
-+ (NSArray *)fileInfoKeys {
-  return @[ kPath, kSHA256, kSHA1, kBundleName, kBundleVersion, kBundleVersionStr,
-            kDownloadReferrerURL, kDownloadURL, kDownloadTimestamp, kDownloadAgent,
-            kType, kPageZero, kCodeSigned, kRule, kSigningChain ];
+// Returns YES if we should output colored text.
+- (BOOL)prettyOutput {
+  return isatty(STDOUT_FILENO) && !self.jsonOutput;
 }
 
-+ (NSArray *)signingChainKeys {
-  return @[ kSHA256, kSHA1, kCommonName, kOrganization, kOrganizationalUnit, kValidFrom,
-            kValidUntil ];
+// Look at environment variable to try to discover user's directory color preference.
+- (NSString *)getDirectoryTTYColor {
+  NSString *lscolors = [[NSProcessInfo processInfo] environment][@"LSCOLORS"];
+  if (!lscolors || lscolors.length < 2) {
+    return @"\033[1;35m"; // bold magenta
+  }
+  char fg = [lscolors characterAtIndex:0];
+  char bg = [lscolors characterAtIndex:1];
+  char validChars[] = "abcdefghxABCDEFGHX";
+  if (!strchr(validChars, fg) || !strchr(validChars, bg)) {
+    return @"\033[1;35m"; // bold magenta
+  }
+  NSMutableString *code = @"\033[".mutableCopy;
+  if (isupper(fg)) {
+    [code appendString:@"1;"];
+    fg = tolower(fg);
+  }
+  if (fg == 'x') {
+    [code appendFormat:@"0"];
+  } else {
+    [code appendFormat:@"%d", fg - 'a' + 30];
+  }
+  if (isupper(bg)) bg = tolower(bg);
+  if (bg != 'x') {
+    [code appendFormat:@";%d", fg - 'a' + 40];
+  }
+  [code appendString:@"m"];
+  return code.copy;
 }
 
-+ (NSString *)printKeyArray:(NSArray *)array {
-  __block NSMutableString *string = [[NSMutableString alloc] init];
-  [array enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
-    [string appendString:[NSString stringWithFormat:@"                       \"%@\"\n", obj]];
-  }];
-  return string;
+// Print out file info for the object at the given path or, if path is a directory and the
+// --recursive flag is set, print out file info for all objects in directory tree.
+- (void)recurseAtPath:(NSString *)path {
+  NSFileManager *fm = [NSFileManager defaultManager];
+  BOOL isDir = NO, isBundle = NO;
```

Comment:
Can you point me to a reference for this?  I know ivars are always initialized to 0, but I didn't think that local variables had any guaranteed initialization, except that maybe under ARC local object vars are initialized to zero.  I couldn't find anything saying otherwise, but I'm not that familiar with new rules under ARC.